ui_advanced_config.py

```diff
@@ -617,291 +628,511 @@ def updateComponentsEntryWidgets(self, v):
-                    self.componentTypeAttr[9]
-                ),
-            )
-        elif (
-            self.componentData[self.componentIDs[currentComponentIdx]].getData("ctype")
-            == "Radio"
-        ):
+                0, currentComp.getData(self.componentTypeAttr[9])
+            )
+        elif currentComp.getData("ctype") == "Radio":
             self.componentsTypeAttr1Label.config(text=self.componentTypeAttr[4])
             self.componentsTypeAttr1Entry.insert(
-                0,
-                self.componentData[self.componentIDs[currentComponentIdx]].getData(
-                    self.componentTypeAttr[4]
-                ),
+                0, currentComp.getData(self.componentTypeAttr[4])
             )
             self.componentsTypeAttr2Label.config(text=self.componentTypeAttr[5])
             self.componentsTypeAttr2Entry.insert(
-                0,
-                self.componentData[self.componentIDs[currentComponentIdx]].getData(
-                    self.componentTypeAttr[5]
-                ),
+                0, currentComp.getData(self.componentTypeAttr[5])
             )
             self.componentsTypeAttr3Label.config(text=self.componentTypeAttr[6])
             self.componentsTypeAttr3Entry.insert(
-                0,
-                self.componentData[self.componentIDs[currentComponentIdx]].getData(
-                    self.componentTypeAttr[6]
-                ),
-            )
-        elif (
-            self.componentData[self.componentIDs[currentComponentIdx]].getData("ctype")
-            == "Arm"
-        ):
+                0, currentComp.getData(self.componentTypeAttr[6])
+            )
+        elif self.currentComponentData.getData("ctype") == "Arm":
             self.componentsTypeAttr1Label.config(text=self.componentTypeAttr[4])
             self.componentsTypeAttr1Entry.insert(
-                0,
-                self.componentData[self.componentIDs[currentComponentIdx]].getData(
-                    self.componentTypeAttr[4]
-                ),
+                0, currentComp.getData(self.componentTypeAttr[4])
             )
             self.componentsTypeAttr2Label.config(text=self.componentTypeAttr[5])
             self.componentsTypeAttr2Entry.insert(
-                0,
-                self.componentData[self.componentIDs[currentComponentIdx]].getData(
-                    self.componentTypeAttr[5]
-                ),
+                0, currentComp.getData(self.componentTypeAttr[5])
             )
             self.componentsTypeAttr3Label.config(text=self.componentTypeAttr[6])
             self.componentsTypeAttr3Entry.insert(
-                0,
-                self.componentData[self.componentIDs[currentComponentIdx]].getData(
-                    self.componentTypeAttr[6]
-                ),
+                0, currentComp.getData(self.componentTypeAttr[6])
             )
 
-    def updateObjectsEntryWidgets(self, v):
```

Comment:
I'm not sure if this the best place for it, but the unsaved changes error (attached) seems to be occurring inconsistently. I can't tell exactly what circumstances makes it occur: it seems to be dependent on clicking on another section (i.e. teams) not switching between objects, and when it does occur, I haven't made a change. This might be accociated with me generally clicking lots of things and making it uphappy though.
<img width="299" alt="Screen Shot 2024-03-02 at 1 07 33 AM" src="https://github.com/Leading-AI-Development-Solutions/MAIA/assets/100381732/ea3387c0-8c12-4331-9bbf-165d02936693">

Reply:
Should be fixed in my next commit.